File: src/main/java/com/anod/appwatcher/search/ResultsAdapterSearch.java
package com.anod.appwatcher.search;

import android.content.Context;

import com.anod.appwatcher.market.SearchEndpoint;
import com.anod.appwatcher.model.WatchAppList;
import com.google.android.finsky.api.model.Document;

/**
 * @author algavris
 * @date 26/08/2016.
 */

public class ResultsAdapterSearch extends ResultsAdapter {
    private final SearchEndpoint mSearchEngine;

    public ResultsAdapterSearch(Context context, SearchEndpoint searchEngine, WatchAppList newAppHandler) {
        super(context, newAppHandler);
        mSearchEngine = searchEngine;
    }

    @Override
    public Document getDocument(int position) {
        boolean isLastPosition = mSearchEngine.getCount() - 1 == position;
        return mSearchEngine.getData().getItem(position, isLastPosition);
    }

    @Override
    public int getItemCount() {
        return mSearchEngine.getCount();
    }

}

File: app/src/main/java/com/anod/appwatcher/search/ResultsAdapterSearch.kt
package com.anod.appwatcher.search

import android.content.Context
import finsky.api.model.Document

/**
 * @author Alex Gavrishev
 * *
 * @date 26/08/2016.
 */

class ResultsAdapterSearch(context: Context, viewModel: SearchViewModel)
    : ResultsAdapter(context, viewModel) {

    private val endpoint = viewModel.endpointSearch

    override fun document(position: Int): Document {
        val isLastPosition = endpoint.count - 1 == position
        return endpoint.data!!.getItem(position, isLastPosition)!!
    }

    override fun getItemCount(): Int {
        return endpoint.count
    }
}
